dask/order.py

```diff
@@ -121,11 +122,40 @@ def order(
 
     leaf_nodes = {k for k, v in dependents.items() if not v}
     root_nodes = {k for k, v in dependencies.items() if not v}
+
+    result: dict[Key, Order | int] = {}
+
+    # Normalize the graph by removing leaf nodes that are not actual tasks, see
+    # for instance da.store where the task is merely an alias
+    # to multiple keys, i.e. [key1, key2, ...,]
+    all_tasks = False
+    j = 0
```

Comment:
nit: `j` could have a more meaningful name like
```suggestion
    n_removed_leaves = 0
```